cmd/image-builder/image-builder.md

```diff
@@ -0,0 +1,170 @@
+It can run in two modes. The first mode is the default mode, where Image Builder acts as a client and triggers the ADO pipeline.
+In this mode, Image Builder supports running as part of a GitHub Actions workflow.
+In the second mode, Image Builder runs as part of the `oci-image-builder` pipeline in the ADO backend.
+The Image Builder is built and pushed as a container image to the Google Artifact Registry repository.
+
+Key features:
+
+* Automatically provides a default tag, which is computed based on a template provided in `config.yaml`
+* Supports adding multiple tags to the image
+* Supports pushing images to Google Artifact Registries.
+* Supports running in a GitHub Actions workflow.
+* Supports building images using the ADO pipeline backend.
+
+
+## Configuration
+
+Image Builder is configured using a global configuration YAML file, set of environment variables, and command line flags.
+
+### Configuration YAML File
+
+`image-builder` requires a configuration YAML file. The file holds the global configuration for the tool and is maintained by the authors.
+Use the `--config` flag to provide a path to the config YAML file.
+
+For more information about available properties in the configuration file, refer to the [config.go](config.go) file.
+
+Here's an example file:
+
+```yaml
+registry: eu.gcr.io/kyma-project
+reproducible: true
+log-format: json
+cache:
+  enabled: true
+  cache-repo: eu.gcr.io/sap-kyma-neighbors-dev/cache
+  cache-run-layers: true
+```
+
+### Environment Variables
+
+Environment variables are mainly used to provide runtime values and configuration set by the CI/CD system.
+They provide details about the context in which the tool is running.
+
+Here is the list of environment variables used by Image Builder:
+
+The Image Builder uses several environment variables, which can be grouped by their use cases as follows:
+
+- `ADO_PAT`: Personal Access Token used to authenticate with the ADO API.
+- `REPO_OWNER`: Used to extract the repository owner for the ADO pipeline.
+- `REPO_NAME`: Used to extract the repository name for the ADO pipeline.
+- `JOB_TYPE`: Determines the type of the job (presubmit or postsubmit).
+- `PULL_NUMBER`: Used when the job type is a presubmit job.
+- `PULL_BASE_SHA`: Used to fetch the base commit SHA for the image tag.
+- `PULL_PULL_SHA`: Used when the job type is a pull request.
+- `CI`: Determines if the Image Builder is running inside a CI system. If set to "true", the CI system is determined and the git state is
+  loaded accordingly.
+- `GITHUB_REPOSITORY`: Used to get the repository name when the CI system is GitHub Actions.
+- `GITHUB_EVENT_NAME`: Used to determine the job type when the CI system is GitHub Actions.
+- `GITHUB_EVENT_PATH`: Used to get the path to the event JSON file when the CI system is GitHub Actions.
+- `GITHUB_SHA`: Used to get the commit SHA when the CI system is GitHub Actions.
+- `GITHUB_REF`: Used to get the pull request number when the CI system is GitHub Actions.
```

Comment:
```suggestion
- **ADO_PAT**: Personal Access Token used to authenticate with the ADO API.
- **REPO_OWNER**: Used to extract the repository owner for the ADO pipeline.
- **REPO_NAME**: Used to extract the repository name for the ADO pipeline.
- **JOB_TYPE**: Determines the type of the job (presubmit or postsubmit).
- **PULL_NUMBER**: Used when the job type is a presubmit job.
- **PULL_BASE_SHA**: Used to fetch the base commit SHA for the image tag.
- **PULL_PULL_SHA**: Used when the job type is a pull request.
- **CI**: Determines if Image Builder is running inside a CI system. If set to "true", the CI system is determined and the git state is
  loaded accordingly.
- **GITHUB_REPOSITORY**: Used to get the repository name when the CI system is GitHub Actions.
- **GITHUB_EVENT_NAME**: Used to determine the job type when the CI system is GitHub Actions.
- **GITHUB_EVENT_PATH**: Used to get the path to the event JSON file when the CI system is GitHub Actions.
- **GITHUB_SHA**: Used to get the commit SHA when the CI system is GitHub Actions.
- `GITHUB_REF**: Used to get the pull request number when the CI system is GitHub Actions.
```

Reply:
Please use [bold font for variables](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#code-font-and-bold-font).